community/en/tiny_imagenet_custom_resnet.ipynb

```diff
@@ -0,0 +1,552 @@
+    "  </td>\n",
+    "</table>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "xHxb-dlhMIzW"
+   },
+   "source": [
+    "## Overview\n",
+    "{TODO: Fill in detailed info of what this accomplishes}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install -q tensorflow==2.0.0-beta1\n",
+    "import tensorflow as tf\n",
+    "assert tf.__version__.startswith('2')\n",
+    "\n",
+    "print(f'{tf.__version__}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "download_path = os.getcwd()\n",
+    "    \n",
+    "path = tf.keras.utils.get_file('tiny-imagenet-200.zip', extract=True, cache_subdir=download_path,\n",
+    "                               origin='http://cs231n.stanford.edu/tiny-imagenet-200.zip')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Image augmentation and image generators"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from tensorflow.keras.preprocessing.image import ImageDataGenerator\n",
+    "\n",
+    "val_data = pd.read_csv(VAL_ANNOT , sep='\\t', names=['File', 'Class', 'X', 'Y', 'H', 'W'])\n",
+    "val_data.drop(['X','Y','H', 'W'], axis=1, inplace=True)\n",
+    "\n",
+    "train_datagen = ImageDataGenerator(\n",
+    "        rescale=1./255,\n",
+    "        rotation_range=18, # Rotation Angle\n",
+    "        zoom_range=0.15,  # Zoom Range\n",
+    "        width_shift_range=0.2, # Width Shift\n",
+    "        height_shift_range=0.2, # Height Shift\n",
+    "        shear_range=0.15,  # Shear Range\n",
+    "        horizontal_flip=True, # Horizontal Flip\n",
+    "        fill_mode=\"reflect\", # Fills empty with reflections\n",
+    "        brightness_range=[0.4, 1.6],  # Increasing/decreasing brightness\n",
+    ")\n",
+    "\n",
+    "train_generator = train_datagen.flow_from_directory(\n",
+    "        TRAIN,\n",
+    "        target_size=(64, 64),\n",
+    "        batch_size=128,\n",
+    "        class_mode='categorical')\n",
+    "\n",
+    "val_datagen = ImageDataGenerator(rescale=1./255)\n",
+    "\n",
+    "val_generator = val_datagen.flow_from_dataframe(\n",
+    "    val_data, directory=VAL, \n",
+    "    x_col='File', \n",
+    "    y_col='Class', \n",
+    "    target_size=(64, 64),\n",
+    "    color_mode='rgb', \n",
+    "    class_mode='categorical', \n",
+    "    batch_size=128, \n",
+    "    shuffle=False, \n",
+    "    seed=42\n",
+    ")"
+   ]
```

Comment:
I'm getting a warning here:

```
UserWarning: Found 10000 invalid image filename(s) in x_col="File". These filename(s) will be ignored.
  .format(n_invalid, x_col)
```

Reply:
This is because the val images are located at `VAL = data_dir/"val/images"` rather than `VAL = data_dir/"val"`. I have fixed it.